acq4/util/generator/waveforms.py

```diff
@@ -320,16 +326,17 @@ def noise(params, mean, sigma, start=0.0, stop=None):
         stop = int(stop * rate)
         
     if stop > nPts-1:
-        params['message'] += "WARNING: Function is longer than generated waveform\n"    
+        warnings.append("WARNING: Function is longer than generated waveform\n")
```

Comment:
Also note that `warnings` is now a list, and any warning messages you generate should be appended to the list.